### PR TITLE
roachtest: add jobs contention roachtest

### PR DESCRIPTION
**Pull request by msbutler**
This patch adds a roachtest that attempts to stress the job system. To do so,
the test spins up 5,000 tables with changefeeds with a 5 minutes pts
expiration, randomly pauses and resume these changefeeds to stress the pts
management poller, and polls SHOW JOBS to assert that it returns in less than 2
minutes. If the assertion fails, the debug zip will hopefully reveal some form
of contention on the job system.  Further, the test also runs EXPLAIN ANALYZE
SHOW JOBS after the assertion fails for more debugging information.

Informs #118512

Release note: none

### CODE REVIEW

**Review by stevendanna — COMMENTED** [review]
Nice.

I would maybe just call this "stress" since we aren't _explicitly_ creating any particular contention, we are just making a bunch of jobs and making sure we can handle them.

I left some comments, but let me know if you want me to carry this forward as I imagine this is lower priority for you atm.

File: pkg/cmd/roachtest/tests/jobs.go [hunk 1]
```diff
@@ -12,188 +12,260 @@ package tests
...
-	// that have data also have a running process.
-	t.Status(fmt.Sprintf("restarting %s (node restart test is done)\n", target))
-	// Don't begin another backup schedule, as the parent test driver has already
-	// set or disallowed the automatic backup schedule.
-	if err := c.StartE(ctx, t.L(), option.DefaultStartOptsNoBackups(),
-		install.MakeClusterSettings(cfg.restartSettings...), target); err != nil {
-		return errors.Wrapf(err, "could not restart node %s", target)
-	}
+	t.L().Printf("SHOW JOBS ran in %.2f seconds", timeutil.Since(queryBegin).Seconds())
+	// TODO(msbutler): once SHOW JOB ID no longer conducts a full scan, test its latency on a random changefeed job.
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/cmd/roachtest/tests/jobs.go]: Here's a Steven caliber mystery:

I try implemented a SHOW JOB latency checker in the latest commit, which requires grabbing a random jobID:
```
 SELECT job_id FROM [SHOW JOBS] ORDER BY random() LIMIT 1
```
But the same jobid gets repeatedly returned in the roachtest.

But in demo, the exact same query returns different job ids, as expected:
```
demo@127.0.0.1:26257/demoapp/movr> SELECT job_id FROM [SHOW JOBS] ORDER BY random() LIMIT 1;
        job_id
----------------------
  945552468363182081
(1 row)

Time: 15ms total (execution 14ms / network 0ms)

demo@127.0.0.1:26257/demoapp/movr> SELECT job_id FROM [SHOW JOBS] ORDER BY random() LIMIT 1;
        job_id
----------------------
  945552471724621825
(1 row)

Time: 12ms total (execution 12ms / network 0ms)
```

Do you see anything in my roachtest code that could explain this?

[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/cmd/roachtest/tests/jobs.go]: Huh, i wasn't patient enough. Eventually, we start to select different job ids....
```
22:04:50 jobs.go:207: SHOW JOB 945582990412709891 ran in 0.01 seconds
22:04:55 jobs.go:161: Created 1000 tables so far
22:05:00 jobs.go:245: Failed to run PAUSE JOB $1 on 0 of 103 jobs
22:05:01 jobs.go:245: Failed to run RESUME JOB $1 on 7 of 103 jobs
22:05:12 jobs.go:207: SHOW JOBS ran in 0.18 seconds
22:05:12 jobs.go:207: SHOW JOB 945582990412709891 ran in 0.01 seconds
22:05:14 jobs.go:245: Failed to run PAUSE JOB $1 on 0 of 119 jobs
22:05:15 jobs.go:245: Failed to run RESUME JOB $1 on 4 of 119 jobs
22:05:31 jobs.go:207: SHOW JOBS ran in 0.27 seconds
22:05:31 jobs.go:207: SHOW JOB 945582990412709891 ran in 0.02 seconds
22:05:33 jobs.go:207: SHOW JOBS ran in 0.21 seconds
22:05:33 jobs.go:207: SHOW JOB 945582990412709891 ran in 0.01 seconds
22:05:42 jobs.go:245: Failed to run PAUSE JOB $1 on 0 of 142 jobs
22:05:43 jobs.go:245: Failed to run RESUME JOB $1 on 11 of 142 jobs
22:06:01 jobs.go:245: Failed to run PAUSE JOB $1 on 0 of 156 jobs
22:06:03 jobs.go:207: SHOW JOBS ran in 0.50 seconds
22:06:03 jobs.go:245: Failed to run RESUME JOB $1 on 15 of 156 jobs
22:06:03 jobs.go:207: SHOW JOB 945583415293968386 ran in 0.01 seconds
22:06:18 jobs.go:245: Failed to run PAUSE JOB $1 on 0 of 166 jobs
22:06:19 jobs.go:207: SHOW JOBS ran in 0.47 seconds
22:06:19 jobs.go:207: SHOW JOB 945583389813342211 ran in 0.01 seconds
```

[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/cmd/roachtest/tests/jobs.go]: Why the heck is `random()` so streaky:
```
22:06:47 jobs.go:207: SHOW JOB 945582990412709891 ran in 0.01 seconds
22:06:58 jobs.go:207: SHOW JOBS ran in 0.19 seconds
22:06:58 jobs.go:207: SHOW JOB 945582990412709891 ran in 0.02 seconds
22:07:02 jobs.go:207: SHOW JOBS ran in 0.32 seconds
22:07:02 jobs.go:207: SHOW JOB 945582990412709891 ran in 0.01 seconds
22:07:03 jobs.go:161: Created 2000 tables so far
22:07:04 jobs.go:245: Failed to run PAUSE JOB $1 on 0 of 199 jobs
22:07:05 jobs.go:245: Failed to run RESUME JOB $1 on 11 of 199 jobs
22:07:25 jobs.go:207: SHOW JOBS ran in 0.32 seconds
22:07:25 jobs.go:207: SHOW JOB 945583685920096257 ran in 0.01 seconds
22:07:33 jobs.go:207: SHOW JOBS ran in 0.30 seconds
22:07:33 jobs.go:207: SHOW JOB 945583685920096257 ran in 0.01 seconds
22:07:36 jobs.go:245: Failed to run PAUSE JOB $1 on 0 of 222 jobs
22:07:38 jobs.go:245: Failed to run RESUME JOB $1 on 11 of 222 jobs
22:07:49 jobs.go:207: SHOW JOBS ran in 0.52 seconds
22:07:49 jobs.go:207: SHOW JOB 945583685920096257 ran in 0.01 seconds
22:07:55 jobs.go:207: SHOW JOBS ran in 0.35 seconds
22:07:55 jobs.go:207: SHOW JOB 945583685920096257 ran in 0.05 seconds
```

[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/cmd/roachtest/tests/jobs.go]: Steven I grant you 30 minutes max to look into this.

[the remark this answers is not in the document]
Comment (stevendanna) [reply pkg/cmd/roachtest/tests/jobs.go]: Nope.

**Review by stevendanna — APPROVED** [review]

File: pkg/cmd/roachtest/tests/jobs.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/cmd/roachtest/tests/jobs.go]: :D

**Comment by msbutler** [comment]
TFTR!

bors r=stevendanna

**Pull request merged**